src/elements/content-sidebar/activity-feed/common/delete-confirmation/DeleteConfirmation.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import { FormattedMessage, MessageDescriptor } from 'react-intl';
+
+import Button from '../../../../../components/button';
+import commonMessages from '../../../../common/messages';
+import PrimaryButton from '../../../../../components/primary-button';
+import { KEYS } from '../../../../../constants';
+import { Overlay } from '../../../../../components/flyout';
+import { ACTIVITY_TARGETS } from '../../../../common/interactionTargets';
+import './DeleteConfirmation.scss';
+
+export interface DeleteConfirmationProps {
+    isOpen: boolean;
+    message: MessageDescriptor;
+    onDeleteCancel: () => void;
+    onDeleteConfirm: () => void;
+    className?: string;
+}
+
+class DeleteConfirmation extends React.Component<DeleteConfirmationProps> {
+    onKeyDown = (event: React.KeyboardEvent): void => {
+        const { isOpen, onDeleteCancel } = this.props;
+
+        event.stopPropagation();
+
+        switch (event.key) {
+            case KEYS.escape:
+                event.stopPropagation();
+                event.preventDefault();
+                if (isOpen) {
+                    onDeleteCancel();
+                }
+                break;
+            default:
+                break;
+        }
+    };
+
+    render(): React.ReactNode {
```

Comment:
You don't need to add the type here.
```suggestion
    render() {
```